Configurations/CommonBase.xcconfig

```diff
@@ -97,17 +97,17 @@ CLANG_CXX_STANDARD_LIBRARY_HARDENING = extensive;
 // Setting CLANG_CXX_STANDARD_LIBRARY_HARDENING is sufficient in recent XCode versions but we want to keep
 // the libcpp assertion enabled on downlevels so use `-D_LIBCPP_ENABLE_ASSERTIONS=1` as well.
 WK_LIBCPP_ASSERTIONS_CFLAGS = $(WK_LIBCPP_ASSERTIONS_CFLAGS_$(WK_PLATFORM_NAME));
-WK_LIBCPP_ASSERTIONS_CFLAGS_iphoneos = $(WK_LIBCPP_ASSERTIONS_CFLAGS$(WK_IOS_17)$(WK_IOS_18));
-WK_LIBCPP_ASSERTIONS_CFLAGS_iphonesimulator = $(WK_LIBCPP_ASSERTIONS_CFLAGS$(WK_IOS_17)$(WK_IOS_18));
+WK_LIBCPP_ASSERTIONS_CFLAGS_iphoneos[sdk=iphone*17*] = $(WK_LIBCPP_ASSERTIONS_CFLAGS$(WK_IOS_17)$(WK_IOS_18));
```

Comment:
For example, this is already checking for target version, is that working incorrectly somehow? I don't think that it will be easy to understand why we use both `[sdk=iphone*17*]` and `WK_IOS_17` checks on the same line.

Besides, we don't support iOS 17 at all, neither as target, nor as SDK.

Reply:
I just applied the same logic from Mac, which was what I was testing, so maybe I should restrict this change to Mac.

My understanding, and it seems to have worked locally, is that `WK_IOS_17` is testing the target, while `[sdk=iphone*17*]` is testing the SDK version being used to compile. Is that not correct?

Reply:
That's likely correct (I didn't check). Are you saying that libcpp assertions require both new SDK and new runtime?